### DISCUSSION TITLE
Unable to set auth token

### DISCUSSION THREAD

**Discussion by Mrcubix** [streamlink/streamlink#5737] [Q&A] [answered]
### Description

Currently having an issue with streamlink not accepting the content of the twitch plugin config file for some reasons.

As per the documentation, i created the neccessary folder & file leading to `$HOME/.config/streamlink/config.twitch`

In this file, as per the documentation, i paste the provided content and replace the example key with my own.

`twitch-api-header=Authorization=OAuth myown32longtwitchauthkeyobtained`

When then launching any streams using the following command:

`streamlink -p mpv -a "--cache=yes" twitch.tv/twitch best --twitch-disable-ads`

I get an error along the like of:

`No plugin can handle URL: --twitch-api-header=Authorization=Oauth myown32longtwitchauthkeyobtained`

As far as i'm aware, i have done it as the documentation indicate, either something changed or i need additionnal argument to make it work while using an auth key.

Would like to know what is wrong so i can update my app and add additional authentication for gql.

**Comment by bastimeyer** [comment]
> No plugin can handle URL: --twitch-api-header

This looks like you've messed up your config file(s), because it's trying to use `--twitch-api-header` as the URL positional argument.

Post the result of
```bash
bash -c 'for cfg in config{,.twitch}; do echo $cfg; cat "${XDG_CONFIG_HOME:-${HOME}/.config}/streamlink/${cfg}"; echo; done'
```
and
```
streamlink --loglevel=debug twitch.tv/... best
```

Make sure to redact your actual OAuth token when pasting, but don't remove the entire CLI argument.

**Reply by Mrcubix** [reply]
Output of the first set of commands:
```log
cat: /home/pi/.config/streamlink/config: No such file or directory

config.twitch
```

Output of the second
```log
streamlink --loglevel=debug twitch.tv/btmc best
usage: streamlink [OPTIONS] <URL> [STREAM]
streamlink: error: unrecognized arguments: twitch.tv/btmc best
```

of course, if i remove `config.twitch`'s content, then it doesn't throw that error.

**Reply by bastimeyer** [reply]
> ```
> cat: /home/pi/.config/streamlink/config: No such file or directory
> 
> config.twitch
> ```

You didn't copy the whole thing because the `config` part at the top is missing, and as you can clearly see, the entire content of `config.twitch` is missing when it executes `cat /home/pi/.config/streamlink/config.twitch`. Fix your `config.twitch` file. It's clearly broken.

**Reply by Mrcubix** [reply]
i didn't feel like it was since i only wanted to configure the twitch plugin, not streamlink as a whole.
I also only needed to add the twitch-api-header arg and adding it to the command itself in bash cause the same set of argument parsing issues:

`streamlink --loglevel=debug "--twitch-api-header=Authorization=OAuth abcdefghijklmnopqrstuvwxyz0123" twitch.tv/... best`
`streamlink --loglevel=debug --twitch-api-header="Authorization=OAuth abcdefghijklmnopqrstuvwxyz0123" twitch.tv/... best`

which result in the same kind of errors.
When specifying the url and quality via `--url` & `--default-stream` respectively, the original error found in my first post can be obtained.


![image](https://github.com/streamlink/streamlink/assets/39861216/1a5250a0-23ae-4cab-846e-4a60f15e8b1f)

Here is the content of config.twitch when the command is run, unless i'm supposed to add something else, i just don't really know what i would need to add, looking at the different doc pages.

**Reply by bastimeyer** [reply]
A screenshot of your text editor doesn't say anything. The file could have invalid encoding and nobody would know...

In case you didn't understand what I was saying earlier: the BASH snippet I've posted was supposed to list the contents of both the `config` and `config.twitch` files using a simple `cat $filename` command, with an `echo $filename` prior to each of the `cat` commands. Your response was missing the file name text of the `config` file (which you didn't copy for some reason), then it showed that the `config` file was missing, which is fine as it doesn't need to exist, but then your output only included the `config.twitch` file name text without the content of `config.twitch` or `cat`'s error message (if this file was also missing). This doesn't make any sense to me, and you should clearify that instead of posting an image of your text editor. Nobody except you knows your system, so if anyone is offering help with explicit commands to debug this, then please follow these questions precisely.

So, since you don't have the `config` file, let's keep it simple and tell us what output of this is:
```sh
file /home/pi/.config/streamlink/config.twitch
cat /home/pi/.config/streamlink/config.twitch
```

Also please post the output of this
```
streamlink --no-config --loglevel=debug
```
because so far, you didn't tell us anything about your OS and the environment Streamlink is run in. How did you install Streamlink? Did you modify any code? Is this the default Twitch plugin, or are you sideloading a custom version? Please answer these questions.

However, if the following doesn't work, then it's definitely an issue with your Streamlink setup, because `--no-config` skips any config files and the plugin argument should work just fine.
```
streamlink --no-config --twitch-api-header "Authorization=OAuth abcdefghijklmnopqrstuvwxyz0123" twitch.tv/... best
```

**Reply by Mrcubix** [reply]
Weird, file didn't appear as changed and contained stuff, prob a visual bug.

Output of your first set of commands:
```
config
cat: /home/pi/.config/streamlink/config: No such file or directory

config.twitch
twitch-api-header=Authorization=OAuth myown32longtwitchauthkeyobtained
```

Output of file:
`/home/pi/.config/streamlink/config.twitch: ASCII text`

output of `streamlink --no-config --loglevel=debug`:
`streamlink: error: unrecognized arguments: --no-config`

**Reply by bastimeyer** [reply, marked as answer]
> streamlink: error: unrecognized arguments: --no-config

Added in `5.5.0`:
https://streamlink.github.io/changelog.html#streamlink-5-5-0-2023-05-05

You are using an ancient version. Update to the latest release.
https://streamlink.github.io/install.html

**Reply by Mrcubix** [reply]
I can't update, Raspi OS is stuck in bulleyes (Debian 11), and updating manually is not recommended, the latest version available is 2.0.0-1

**Reply by bastimeyer** [reply]
- https://streamlink.github.io/install.html#package-availability
- https://streamlink.github.io/install.html#linux-appimages

Alternatively

- https://streamlink.github.io/install.html#pypi-package-and-source-code
- https://streamlink.github.io/install.html#virtual-environment

**Reply by Mrcubix** [reply]
Seems like it work after installing it via python, makes you think twice before using the OS package manager to install a dependency that's probably outdated.
Thanks !
